spark/src/main/java/org/apache/iceberg/spark/SparkUtil.java

```diff
@@ -100,4 +107,11 @@ public static void validatePartitionTransforms(PartitionSpec spec) {
       }
     }
   }
+
+  public static boolean isLocalityEnabledDefault(Map<String, String> tableProperties, String fsScheme) {
+    String tableLocalityProp = PropertyUtil.propertyAsString(tableProperties, TableProperties.LOCALITY_ENABLED,
+        TableProperties.LOCALITY_ENABLED_DEFAULT);
+    return tableLocalityProp == null ? LOCALITY_WHITELIST_FS.contains(fsScheme) :
+        Boolean.parseBoolean(tableLocalityProp);
```

Comment:
I think that locality should be enabled by default, but only relevant if the file system is white-listed. That would make the logic here `localityEnabled && LOCALITY_WHITELIST_FS.contains(fsScheme)`. That's a bit simpler than using the table property to override and we don't want to check locality for file systems that don't support it.

Reply:
OK. I am using spark-sql and didn't find a way to set `locality` as a spark read option. I will try to add it as a user-defined spark configuration so that it can be set per session.

Reply:
We should add the ability to set read options in hints via Spark extensions. Being able to use hints would be great because we could use it for this and a few other things, like time travel in SQL. (@RussellSpitzer, what do you think?)

Reply:
I'm a big fan of hints (not that I was asked). But they're a great SQL first solution to a number of problems.

Assuming you mean like
```
select /* read.locality=false */ a, b, c from table iceberg_table t
```.

We have encountered a few situations internally where updating `spark.sql.partitions` would make a job unreasonably slow (due to possibly a large filter prior to the write), so a `COALESCE` hint would be really helpful (which I assume is naturally supported unless Iceberg operations get in the way, e.g maybe anything to do with merging or deletes). But I have admittedly not tried it.

Reply:
I think hints could solve a number of problems that people want to create table level properties for, for which table level properties are likely not the most appropriate as they're dependent on seasonal changes in dataset size and also cluster resources when the job is started.

Reply:
Yeah, I'm not a fan either, but the read and write options probably won't be possible through SQL otherwise. Maybe that's not what we want to do for things that would ideally have SQL clauses (like `AS OF TIMESTAMP` or `AS OF VERSION`) but hints like `locality=true` seem like a reasonable path to me. We may even be able to get that in upstream Spark.

Reply:
Lastly, I think hints are much easier to explain to end users and have them remember, as there are A LOT of spark configurations (by necessity) and even people who are very well acquainted with spark can get them mixed up.... which leads to support requests etc.

If the hint gets placed in the query (SQL file, notebook), developers are much more likely to remember what needs to be updated to change it to suit their current needs.

Reply:
I thought for time travel and such I'm a bigger fan of just adding more special table naming, just so we have less reliance on catalyst. That said I have no problem with doing some more hints for things like locality